### PR TITLE
Online Food Ordering Example

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tsurdilo@redhat.com>

**Many thanks for submitting your Pull Request :heart:!**

**Please specify parts this PR updates:**

- [ ] Specification
- [ ] Schema
- [x ] Examples
- [ ] Extensions
- [ ] Roadmap
- [ ] Use Cases
- [ ] Community
- [ ] TCK
- [ ] Other

**What this PR does / why we need it**:
Adds a new example - online food ordering
**Special notes for reviewers**:

**Additional information (if needed):**

### CODE REVIEW

**Comment by tsurdilo** [comment]
parsed markdown for example: https://github.com/tsurdilo/specification/tree/orderdeliveryexample/examples#online-food-ordering

File: examples/README.md [hunk 1]
```diff
@@ -3873,3 +3874,249 @@ states:
 </td>
 </tr>
 </table>
+
+### Online Food Ordering
+
+#### Description
+
+In this example we want to create an online food ordering workflow. Below image outlines the workflow 
+structure and the available services:
```
Comment (ricardozanini) [inline examples/README.md:3883]:
```suggestion
In this example, we want to create an online food ordering workflow. The below image outlines the workflow 
structure and the available services:
```

File: examples/README.md [hunk 2]
```diff
@@ -3873,3 +3874,249 @@ states:
...
+
+Our workflow needs to communicate with three services during its execution, namely the Order, Delivery, and 
+the Payment services. 
+
+For the sake of the example we assume that our workflow can communicate to the Order and Delivery services via REST
+and the Payment service via gRPC. 
+
+Let's start off by defining an example CloudEvent which triggers an instance of our workflow.
+This event can be sent by a web UI for example, or be pushed onto a Kafka/MQTT topic to start 
+our order workflow.
```
Comment (ricardozanini) [inline examples/README.md:3904]:
```suggestion
Our workflow starts with the "Place Order" [Subflow](../specification.md#SubFlow-State), which is responsible
to send the received order to the requested restaurant and the estimated order ETA. 
We then wait for the ETA time when our workflow should go into the "Deliver Order" SubFlow, responsible
for dispatching a Courier and sending her/him off to pick up the order. Once the order is picked up, the Courier needs to deliver the order to the customer. 
After the order has been delivered to the customer, our workflow needs to charge the customer. 

Our workflow needs to communicate with three services during its execution, namely the Order, Delivery, and 
the Payment services. 

For the sake of the example, we assume that our workflow can communicate to the Order and Delivery services via REST and the Payment service via gRPC. 

Let's start by defining an example CloudEvent which triggers an instance of our workflow.
This event can be sent by a web UI, for example, or be pushed onto a Kafka/MQTT topic to start our order workflow.
```

File: examples/README.md [hunk 3]
```diff
@@ -3873,3 +3874,249 @@ states:
...
+         "instructions": ""
+      }
+   }
+}
+```
+
+Note the "orderid" CloudEvent context attribute which contains the unique ID of the order specified 
+in this event. [Event correlation](../specification.md#Correlation-Definition) is done against CE context attributes, and as such to be able
+to correlate multiple order events to the same order id, it needs to be part of the CE context attributes, and 
+not its data (payload).
```
Comment (ricardozanini) [inline examples/README.md:3943]:
```suggestion
Note the `orderid` CloudEvent context attribute, which contains the unique ID of the order specified in this event. [Event correlation](../specification.md#Correlation-Definition) is done against CE context attributes, and as such, to be able
to correlate multiple order events to the same order id, it needs to be part of the CE context attributes, and 
not its data (payload).  
```

File: examples/README.md [hunk 4]
```diff
@@ -3873,3 +3874,249 @@ states:
...
+```
+
+Note the "orderid" CloudEvent context attribute which contains the unique ID of the order specified 
+in this event. [Event correlation](../specification.md#Correlation-Definition) is done against CE context attributes, and as such to be able
+to correlate multiple order events to the same order id, it needs to be part of the CE context attributes, and 
+not its data (payload).  
+
+Now let's start defining our workflow. For the sake of this example let's define our function and event definitions
+as separate YAML files (and then reference them inside our workflow definition). This is useful in cases
+when you want to reuse them between multiple workflow definitions.
```
Comment (ricardozanini) [inline examples/README.md:3947]:
```suggestion
Now let's start defining our workflow. For the sake of this example, let's define our function and event definitions
as separate YAML files (and then reference them inside our workflow definition). This is useful in cases
when you want to reuse them between multiple workflow definitions.
```

File: examples/README.md [hunk 5]
```diff
@@ -3873,3 +3874,249 @@ states:
...
+    - Order Delievered Event
+    eventDataFilter:
+      data: "${ .data.status }"
+      toStateData: "${ .status }"
+  end: true
+```
+
+#### Workflow Results
+
+For the example order event the workflow output for a successful completion would look like for example:
```
Comment (ricardozanini) [inline examples/README.md:4108]:
```suggestion
For the example order event, the workflow output for a successful completion would look like for example:
```

**Review by ricardozanini — APPROVED** [review]

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**